doc/Caliptra.md

```diff
@@ -252,11 +252,15 @@ The following figure shows the basic high-level blocks of Caliptra.
 
 *Figure 2: Caliptra high level blocks*
 
-![](./images/Caliptra_HW_diagram.png)
+![](./images/Caliptra2p0.png)
 
 See the [hardware section](#hardware) for a detailed discussion.
 
-Caliptra is among the first uncore microcontrollers taken out of reset by the power-on reset logic. Caliptra coordinates the start of the firmware chain-of-trust with the immutable component of the SoC ROM. After the Caliptra ROM completes initialization, it provides a "stash measurement" API and callback signals for the SoC ROM to proceed with the boot process. Caliptra ROM supports stashing of at most eight measurements prior to the boot of Caliptra firmware. The SoC then may choose to boot Caliptra firmware. Any security-sensitive code or configuration loaded by the SoC prior to Caliptra firmware boot must be stashed within Caliptra. If the SoC exceeds Caliptra ROM's measurement stash capacity, attestation must be disabled until the next cold reset. The boot process is as follows:
+From Caliptra 2.x onwards, Caliptra introduces two modes of operation. **Passive** mode which was supported in 1.x architecture and **Subsystem** mode. Fundamental difference between passive mode and subsystem mode is that in the subsystem mode Caliptra is the RoT for the SOC and provides streaming boot, secure boot and attestation. In Subsystem mode, Caliptra also provides various crypto API services such as encryption/decryption of SOC FWs, Key releases, Key wraps, hashing etc. to name a few. Please see Caliptra subsystem mode Crypto API section for more details (**FIXME**: section name & details).
```

Comment:
I think in passive mode many of these things are true as well. I would say something like this instead

```suggestion
From Caliptra 2.x onwards, Caliptra introduces two modes of operation. **Passive** mode which was supported in 1.x architecture and **Subsystem** mode.

In sybststem mode, Caliptra is the only RoT for the SoC, and provides the following additional services:

* Streaming boot
* Secure boot enforcement
* SoC FW Decryption

In both modes, Caliptra 2.0 provides additional crypto offload features such as key release, key wrapping, hashing etc. Please see Caliptra subsystem mode Crypto API section for more details (**FIXME**: section name & details).
```

Reply:
Hi Jordan - For key wrapping etc. functions & key releases, you need AXI manager to be operational as Caliptra directly writes to the destination; Since passive only has AXI-SUB, crypto offload functions are not possible without FW having visibility to keys; even hashing is mailbox size dependent which subsystem doesnt constrain

Reply:
* For hashing, I don't think there's significant demand for mailbox hashing in passive mode. I think we'd mainly be looking for streaming mode support.
* For key wrapping, I need to understand this a bit better. I'll do another pass through the spec. But I'll consider that non-blocking for this PR.

So my main suggestions for this text are mostly wording changes at this point. I want to make it clear that subsystem is additive and that passive mode is still fully supported.

I would be ok with just opening an issue and I can make a PR later with suggestions.

Reply:
" I think we'd mainly be looking for streaming mode support." -> my take (or strong stance) is if someone wants to Caliptra to act as security controller use Caliptra in subsystem mode.

"For key wrapping, I need to understand this a bit better." -> There is no hardware in passive mode to take the wrapped key and put it in MB or register for SOC to read. Not a HW feature. If FW can see the wrapped keys, then you can do this.

Reply:
Also generally speaking, there was no direction/decision to make HW between passive and active to act the same way (and there is too much rearchitecture to do now given the freeze is end of this month). The only thing with passive was to keep legacy (FW) capabilities intact. And the only one we fixed/changed is SHA streaming boot which was a FIPS issue and functional problem in 1.0